packages/story-editor/src/components/panels/shared/media/fileRow.js

```diff
@@ -0,0 +1,183 @@
+  width: 24px;
+  height: 24px;
+  vertical-align: bottom;
+`;
+
+const UploadIcon = styled(Icons.ArrowOutline)`
+  width: 24px;
+  height: 24px;
+  vertical-align: bottom;
+`;
+
+const Link = styled.a`
+  ${({ theme }) => css`
+    color: ${theme.colors.fg.primary};
+    text-decoration: none;
+    cursor: pointer;
+
+    :hover,
+    :active,
+    :focus {
+      text-decoration: underline;
+      /* Override WordPress's common css */
+      color: ${theme.colors.fg.primary} !important;
+    }
+
+    ${themeHelpers.focusableOutlineCSS(theme.colors.border.focus)}
+  `};
+`;
+
+const StyledButton = styled(Button)`
+  ${({ theme }) =>
+    themeHelpers.focusableOutlineCSS(
+      theme.colors.border.focus,
+      theme.colors.bg.secondary
+    )};
+`;
+
+const FileName = styled(Text).attrs({
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL,
+})`
+  color: ${({ theme }) => theme.colors.fg.primary};
+`;
+
+function FileRow({
+  id,
+  isExternal,
+  src,
+  title,
+  options,
+  removeItemLabel,
+  onRemove,
+  children,
+}) {
+  const optionsWithGroup = [
+    {
+      group: options,
+    },
+  ];
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const onMenuOpen = () => setIsMenuOpen(true);
+  const onMenuClose = () => setIsMenuOpen(false);
+  const onMenuSelected = useCallback(
+    (value) => {
+      setIsMenuOpen(false);
+      options.find((item) => item.value === value)?.onClick();
+    },
+    [options]
+  );
+
+  return (
+    <Row key={`row-filename-${id}`}>
+      <Tooltip
+        title={
+          isExternal
+            ? __('External file', 'web-stories')
+            : __('Local file', 'web-stories')
+        }
+      >
+        {isExternal ? <LinkIcon /> : <UploadIcon />}
+      </Tooltip>
+      <InputRow>
+        <FileName>
+          <Link href={src} rel="noreferrer" target="_blank">
+            {title}
+          </Link>
+        </FileName>
+      </InputRow>
+      {children}
+      {options.length > 0 && (
+        <Tooltip title={__('Replace', 'web-stories')}>
+          <DropDownMenu
+            onMenuOpen={onMenuOpen}
+            isMenuOpen={isMenuOpen}
+            onMenuSelected={onMenuSelected}
+            display
+            onMenuClose={onMenuClose}
+            options={optionsWithGroup}
+          >
+            <ReplaceIcon />
+          </DropDownMenu>
+        </Tooltip>
+      )}
+      <Tooltip hasTail title={removeItemLabel}>
+        <StyledButton
+          aria-label={removeItemLabel}
+          type={BUTTON_TYPES.TERTIARY}
+          size={BUTTON_SIZES.SMALL}
+          variant={BUTTON_VARIANTS.SQUARE}
+          onClick={() => onRemove(id)}
+        >
+          <Icons.Trash />
+        </StyledButton>
+      </Tooltip>
+    </Row>
+  );
+}
+
+FileRow.propTypes = {
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  isExternal: PropTypes.bool,
+  src: PropTypes.string,
+  title: PropTypes.string,
+  options: PropTypes.array,
+  removeItemLabel: PropTypes.string,
```

Comment:
Not the right line of code but it would be nice to announce (speak) that the file is removed for a11y.